promql/promqltest/test.go

```diff
@@ -447,18 +508,35 @@ func (t *test) parseEval(lines []string, i int) (int, *evalCmd, error) {
 		}
 		metric, vals, err := parseSeries(defLine, i)
 		if err != nil {
+			literal, err1 := parseAsStringLiteral(defLine)
+			if err1 == nil {
+				cmd.expectedString = literal
+				cmd.excludeFromRangeQuery = true
+				return i, cmd, nil
+			}
+
 			return i, nil, err
 		}
 
-		// Currently, we are not expecting any matrices.
-		if len(vals) > 1 && isInstant {
+		// Currently, we only allow a range vector for an instant query is where we have defined the expected range vector timestamps.
+		if len(vals) > 1 && isInstant && !allowExpectedRangeVector {
 			return i, nil, formatErr("expecting multiple values in instant evaluation not allowed")
```

Comment:
Should we change this error message to make it clearer that this is supported, but the test author needs to use the `expect range vector` syntax?